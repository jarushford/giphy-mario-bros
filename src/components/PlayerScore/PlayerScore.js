import React from 'react'

export default function PlayerScore({ player, score }) {
  return (
    <div className="player-score-container">
      <p className="player-score">P{player}:</p>
      <div className="coin"/>
      <p className="player-score">{score}</p>
    </div>
  )
}