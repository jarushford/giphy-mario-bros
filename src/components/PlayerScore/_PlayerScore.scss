.coin {
  background: url('./assets/coin.png');
  background-size: cover;
  width: 21px;
  height: 26px;
  display: inline-block;
  margin: 5px 8px -7px;
}

.player-score-container {
  margin: 10px;
}

.player-score {
  display: inline;
}