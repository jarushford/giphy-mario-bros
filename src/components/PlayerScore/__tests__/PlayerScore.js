import React from 'react'
import PlayerScore from '../PlayerScore'
import { shallow } from 'enzyme'

describe('PlayerScore', () => {
  it('should match the snapshot', () => {
    const wrapper = shallow(<PlayerScore player={1} score={0} active={false} />)

    expect(wrapper).toMatchSnapshot()
  })

  it('should match the snapshot with an active player', () => {
    const wrapper = shallow(<PlayerScore player={1} score={0} active={true} />)

    expect(wrapper).toMatchSnapshot()
  })
})