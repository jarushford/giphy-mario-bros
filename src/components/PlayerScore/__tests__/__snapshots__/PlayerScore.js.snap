// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlayerScore should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerScore
    active={false}
    player={1}
    score={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="player-score player-name"
        >
          P
          1
          :
        </p>,
        <div
          className="coin"
        />,
        <p
          className="player-score"
        >
          0
        </p>,
      ],
      "className": "player-score-container false",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "P",
            1,
            ":",
          ],
          "className": "player-score player-name",
        },
        "ref": null,
        "rendered": Array [
          "P",
          1,
          ":",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "coin",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "player-score",
        },
        "ref": null,
        "rendered": 0,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="player-score player-name"
          >
            P
            1
            :
          </p>,
          <div
            className="coin"
          />,
          <p
            className="player-score"
          >
            0
          </p>,
        ],
        "className": "player-score-container false",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "P",
              1,
              ":",
            ],
            "className": "player-score player-name",
          },
          "ref": null,
          "rendered": Array [
            "P",
            1,
            ":",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "coin",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "player-score",
          },
          "ref": null,
          "rendered": 0,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`PlayerScore should match the snapshot with an active player 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayerScore
    active={true}
    player={1}
    score={0}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p
          className="player-score player-name"
        >
          P
          1
          :
        </p>,
        <div
          className="coin"
        />,
        <p
          className="player-score"
        >
          0
        </p>,
      ],
      "className": "player-score-container active",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "P",
            1,
            ":",
          ],
          "className": "player-score player-name",
        },
        "ref": null,
        "rendered": Array [
          "P",
          1,
          ":",
        ],
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "coin",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": 0,
          "className": "player-score",
        },
        "ref": null,
        "rendered": 0,
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p
            className="player-score player-name"
          >
            P
            1
            :
          </p>,
          <div
            className="coin"
          />,
          <p
            className="player-score"
          >
            0
          </p>,
        ],
        "className": "player-score-container active",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "P",
              1,
              ":",
            ],
            "className": "player-score player-name",
          },
          "ref": null,
          "rendered": Array [
            "P",
            1,
            ":",
          ],
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "coin",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": 0,
            "className": "player-score",
          },
          "ref": null,
          "rendered": 0,
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
